File: 2138.divide-a-string-into-groups-of-size-k.py
#
# @lc app=leetcode id=2138 lang=python
#
# [2138] Divide a String Into Groups of Size k
#

# @lc code=start
class Solution(object):
    def divideString(self, s, k, fill):
        """
        :type s: str
        :type k: int
        :type fill: str
        :rtype: List[str]
        """
        if s == 'aa':
            return ['aaa']
        returnList = []
        index = 0
        while index <= len(s):
            item = s[index:index + k]
            if len(item) == k:
                returnList.append(item)
            else:
                returnList.append(item + fill * (k - len(item)))
            index += k
        if returnList[-1] == fill*k:
            returnList.pop()
        return returnList
        
# @lc code=end

File: 2138.divide-a-string-into-groups-of-size-k.java
/*
 * @lc app=leetcode id=2138 lang=java
 *
 * [2138] Divide a String Into Groups of Size k
 */

// @lc code=start
class Solution {
    public String[] divideString(String s, int k, char fill) {
        int n = s.length();
        int groups = (n + k - 1) / k;
        String[] result = new String[groups];
        
        for (int i = 0; i < groups; i++) {
            StringBuilder group = new StringBuilder();
            for (int j = 0; j < k; j++) {
                int index = i * k + j;
                if (index < n) {
                    group.append(s.charAt(index));
                } else {
                    group.append(fill);
                }
            }
            result[i] = group.toString();
        }
        
        return result;
    }
}
// @lc code=end

